### DISCUSSION TITLE
使用复合主键有哪些已知的问题

### DISCUSSION THREAD

**Discussion by hunao0221** [zuihou/zuihou-admin-cloud#316] [企业版]
- 无法自动填充id值

**Comment by zuihou** [comment]
https://baomidou.com/pages/f84a74/#%E5%85%B3%E4%BA%8E-mp-%E4%B8%BA%E4%BD%95%E4%B8%8D%E6%94%AF%E6%8C%81%E5%A4%8D%E5%90%88%E4%B8%BB%E9%94%AE-%E5%BC%BA%E5%88%B6%E5%94%AF%E4%B8%80-id-%E9%97%AE%E9%A2%98%E8%A7%A3%E7%AD%94




关于 MP 为何不支持复合主键，强制唯一 ID 问题解答
为了省一个 ID 字段使用复合主键个人觉得缺点大于优点，不建议也不推荐这么做。

- 增加了表与表之间的相互依赖性
- 增加了数据复杂的约束、规则（约束唯一性，完全可以使用联合索引）
- 增加了更新数据的限制
- 严重的数据冗余和更新异常问题
- 性能问题，查询某个ID时无法使用索引、更新性能下降
